javascript/javascript1/week3/smartApps.js

```diff
@@ -0,0 +1,85 @@
+const activities = [];
+let date = new Date();
+let day = date.getDate();
+let months = date.getMonth() + 1;
+let year = date.getFullYear();
+
+function getSum(arr) {
+  let counter = arr.reduce((a, b) => {
+    return a + b
+  })
+  return counter
+}
+
+function getTime(time) {
+  let hours = Math.floor(time / 60)
+  let minuts = time % 60
+  const timeArr = [time, hours, minuts]
+  return timeArr
+}
+
+
+function addActivity(activity, duration) {
+  if (typeof activity != 'string' || typeof duration != 'number')
+    throw new Error('Illegal Argument Exception');
+
+  let date = `${day}-${months}-${year}`
+  const obj = { 'date': `${date}`, 'activity': `${activity}`, 'duration': duration }
+  activities.push(obj)
+  return obj
+}
+
+function showStatus(date) {
+  if (activities.length === 0) {
+    return 'Add some activities before calling showStatus';
+  }
+  // check if date is there 
+  const filteredArr = activities.filter(obj => obj.date === date)
+  if (filteredArr.length === 0) {
+    console.log('no activities at this date')
+    return;
+  }
+  const mapped = filteredArr.map(obj => obj.duration)
+  // number of activity 
+  const numOfActivities = filteredArr.length;
+  if (numOfActivities > 0) {
+    let sumOfTime = getSum(mapped);
+    let time = getTime(sumOfTime);
+
+    if (sumOfTime > 240) {
+      return 'You have reached your limit, no more smartphoning for you!'
+    }
+    if (60 < sumOfTime && sumOfTime <= 240) {
+      return `You have added ${numOfActivities} activities.They amount to ${time[1]} h and ${time[2]} min of usage`
+    }
+    return `You have added ${numOfActivities} activities.They amount to ${time[0]} min of usage`
+  }
+}
+
+
+function mostTime(array) {
```

Comment:
When a function returns some value to be used, it is common to call it a getter. So this function could be called getMostTimeConsumingActivity.